frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/generate-changed-properties.ts

```diff
@@ -17,7 +16,8 @@ export function generateChangedProperties<Element extends ImmutableJsonObject>(
     const changedProperties = new Set<keyof Element>();
     // add the changed properties to the Set
     for (const property of properties) {
-        if (oldElement[property] !== newElement[property]) {
+        // It is potentially much more efficient to deep compare here instead of having to potentially rerender something unnecessarily
+        if (!isEqual(oldElement[property], newElement[property])) {
```

Comment:
Previously I assumed that we would (nearly) always only change properties that then also have real changes.

Now, because of the time travel it could be, that the same object from two different cached states would end up in here. -> Resulted in recreation of the feature instead of smooth animation to e.g. new position